docs/user-guide/carbon-budget.ipynb

```diff
@@ -0,0 +1,179 @@
+    "\n",
+    "```bash\n",
+    "python -m muse --model default --copy PATH/TO/COPY/THE/MODEL/TO\n",
+    "```\n",
+    "\n",
+    "## Adding a carbon budget\n",
+    "\n",
+    "Using a carbon budget allows the carbon price to be escalated (if necessary), to encourage investment in low-carbon technologies and bring emissions down in line with the budget. To achieve this, MUSE must solve the system with a number of different carbon prices in every year to uncover the relationship between carbon price and emissions. This will most likely be a monotonically decreasing function (i.e. lower emissions as carbon price increases), but may be highly non-linear and discontinuous. By building a picture of this landscape by repeatedly sampling different carbon prices, MUSE can select the carbon price that gives emissions as close as possible to the budget.\n",
+    "\n",
+    "Naturally, this means that the simulation will take much longer to run. Therefore, to ensure that the model runs in a reasonable amount of time, will will first adjust the time framework to run only until 2035:\n",
+    "\n",
+    "```toml\n",
+    "time_framework = [2020, 2025, 2030, 2035]\n",
+    "```\n",
+    "\n",
+    "Then, we will add the following to the settings file to impose a carbon budget:\n",
+    "\n",
+    "```toml\n",
+    "[carbon_budget_control]\n",
+    "budget = [300, 300, 300, 300]\n",
+    "commodities = ['CO2f']\n",
+    "control_undershoot = false\n",
+    "control_overshoot = false\n",
+    "method = 'bisection'\n",
+    "method_options.sample_size = 5\n",
+    "method_options.tolerance = 0.2\n",
+    "```\n",
+    "\n",
+    "We need a budget value for each year in the time framework. In this case we will impose a budget of 300 in every year. `control_undershoot` and `control_overshoot` allow a carbon surplus or deficit in one year to be carried over to the next year (which we will not allow in this case). We also need to give the name of the carbon commodity (matching a commodity defined in `GlobalCommodities.csv`), in this case `CO2f`. \n",
+    "\n",
+    "The other lines indicate settings used in the carbon budget algorithm, which modifies the carbon price so that carbon emissions come as close as possible to the budget. The method we are using is the [bisection method](https://en.wikipedia.org/wiki/Bisection_method), which iteratively halves a carbon price interval until emissions are within tolerance of the target. `sample_size` and `tolerance` are parameters used in the bisection algorithm: `tolerance` controls how close emissions need to be to the budget (e.g. 20%) for the algorithm to converge on a solution; `sample_size` is the maximum number of carbon prices that will be tested. Reducing `tolerance` and increasing `sample_size` can help achieve emissions as close as possible to the budget, but the simulation may take much longer to run.\n",
```

Comment:
The meaning you state here for the `sample_size` is the one for the `fitting` algorithm. For the `bisection` method, it is (taken from the docstrings) "Maximum number of iterations for bisection". I think that we should either use a different parameter (`sample_size` for `fitting` and `steps` for `bisection`, for example) or very clearly explain what `sample_size` represents depending on what algorithm you are using.